### PR TITLE
Expose Proxy feature

### PR DESCRIPTION
**Pull request by stevenschlansker**
This really would be nice: https://java.net/jira/browse/JAX_RS_SPEC-496

### CODE REVIEW

**Comment by pennello** [comment]
The Travis CI tests are failing due to:

    Detected Maven Version: 3.2.5 is not in the allowed range [3.3.3,).

File: client/src/main/java/com/opentable/jaxrs/JaxRsClientFactory.java [hunk 1]
```diff
@@ -213,4 +214,17 @@ public Client newClient(String clientName, JaxRsFeatureGroup feature, JaxRsFeatu
     public Client newClient(String clientName, Collection<JaxRsFeatureGroup> featureGroups) {
         return newBuilder(clientName, featureGroups).build();
     }
+
+    /**
+     * Create a Client proxy for the given interface type.
+     * Note that different JAX-RS providers behave slightly
+     * differently for this feature.
```
Comment (pennello) [inline client/src/main/java/com/opentable/jaxrs/JaxRsClientFactory.java]: Would it be useful here to link to any other documentation on the web, or elsewhere, to give more context about the different JAX-RS providers?

File: clientfactory-jersey/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java [hunk 2]
```diff
@@ -41,6 +42,11 @@ public ClientBuilder newBuilder(String clientName, JaxRsClientConfig config) {
         return builder.register(GZipEncoder.class);
     }
 
+    @Override
+    public <T> T createClientProxy(Class<T> proxyType, WebTarget baseTarget) {
+        throw new UnsupportedOperationException("Jersey support for this feature is TODO");
+    }
+
```
Comment (pennello) [inline clientfactory-jersey/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java]: Heh.  Do we have anyone still using Jersy?  It already doesn't support all of the config options we have for RESTEasy.

Comment (stevenschlansker) [inline clientfactory-jersey/src/main/java/com/opentable/jaxrs/JaxRsClientFactoryImpl.java]: YAGNI.  I think Olivier uses it but he's not exactly following the bleeding edge of our libraries.

**Pull request merged**